### ISSUE TITLE
beamer with --incremental has duplicate page

### ISSUE THREAD

**Issue by Jmuccigr** [jgm/pandoc#4390]
beamer output when the -i flag is set and a list is followed by a pause generates a duplicate of the last complete-list slide. That is, the list completes and then that slide is duplicated. So the short slide show here generates 4, not 3 slides, where slides 2 and 3 are identical.

```
- Ad vestibulum ornare fugiat
- consequat fringilla lacinia 

. . . 

nibh habitasse
```

**Comment by jgm** [comment]
Pandoc's beamer output for this is just
```latex
\begin{itemize}[<+->]
\tightlist
\item
  Ad vestibulum ornare fugiat
\item
  consequat fringilla lacinia
\end{itemize}

\pause

nibh habitasse
```
This is the obvious translation into beamer of the idea of an incremental list followed by a pause -- so maybe this is really a beamer issue?  But if you have thoughts about alternative beamer output that would work, let us know.

**Comment by Jmuccigr** [comment]
Could be beamer, because it definitely inserts a duplicate slide here, but without the pause, the text appears right away. I'll check over there.

**Comment by Jmuccigr** [comment]
OK, this seems to be a known thing. See [here](https://www.texdev.net/2014/01/28/the-overlay-syntax-and-pause-in-beamer/) for the solution:

Don't use `\pause` on slide with other dynamic content, like an incremental list. Instead use `\onslide<+->`.

Simply replacing one with the other makes my case work as expected.

**Comment by Jmuccigr** [comment]
If I understand section 9 of the [beamer manual](http://www.texdoc.net/texmf-dist/doc/latex/beamer/doc/beameruserguide.pdf) correctly, `\onslide<+->` can replace `\pause` wherever it's used. The manual says this about `\pause`:

> A much more fine-grained control over what is shown on each slide can be attained using overlay specifications, see the next sections.

I don't build pandoc myself, but it should be fairly easy to test this. I think the only thing that needs to be changed is one line in the latex writer, 545:

```haskell
then blockToLaTeX (RawBlock "latex" "\\pause")
```
to
```haskell
then blockToLaTeX (RawBlock "latex" "\\onslide<+->")
```

**Comment by jgm** [comment]
If you want to test yourself, you can use this lua filter:
```
function Para(el)
  if pandoc.utils.stringify(el) == ". . ." then
      return pandoc.RawBlock("latex", "\\onslide<+->")
  end
end
```
Save as `onslide.lua` and use with `pandoc -t beamer --lua-filter onslide.lua`.
If you are right that we can just use `\onslide<+->` everywhere instead of `\pause`, I can make the change to the writer.  But, can you test first?

**Comment by Jmuccigr** [comment]
Happy to. Is there a test suite to use?

**Comment by Jmuccigr** [comment]
I haven't been able to figure this out on my own and my question on tex.stackexchange.com hasn't been answered either.

I'm not sure how difficult this would be, but it seems like using \onslide for a pause after a list would work, leaving the \pause in when there hasn't been a list yet. I have no idea how difficult this would be to implement.

**Comment by Jmuccigr** [comment]
One week later, and still no responses to my various queries on stackexchange and tex stackexchange, so I think the only solution is what I suggested last week: don't use `\pause` after a list; use `\onslide<+->` instead.

**Comment by mardukbp** [comment]
@jgm I am having the same issue. What is needed is a syntax that Pandoc parses to `\onslide<+->`. What about `, , ,`?

**Comment by christophehenry** [comment]
This issue feels very disturbing. Is there any reason why `. . .` couldn't translate to `\onslide<+->` instead of `\pause`?

**Comment by jgm** [comment]
> Is there any reason why . . . couldn't translate to \onslide<+-> instead of \pause?

No reason. That would be a sensible change; it's just that we developers don't always have time to make all the sensible changes.

**Comment by jgm** [comment]
OK, i just tested using `\onslide<+->` everywhere instead of `\pause`, and that does not work. No pause is created e.g. between two regular paragraphs.

If you want to push this issue forward, I'd suggest generated a tex file with pandoc that puts pauses in various contexts (both after and not after incremental lists, for example), and then make changes to the tex and recompile to find something that works.  In principle we could give different treatment to `. . .` if it comes after an incremental section, but the code would be more complicated.

**Comment by Jmuccigr** [comment]
It's been a while since I have thought about this, but a slightly younger me said above:

> ...I think the only solution is what I suggested last week: don't use \pause after a list; use \onslide<+-> instead.

So it's not a universal replacement, but a contextual one (= after a list).

**Comment by christophehenry** [comment]
> So it's not a universal replacement, but a contextual one (= after a list).

This is not sufficient. You need to use `\onslide<+->` throughout the whole slide after a list. Using it only once after the list only move the problem to the next pause. Here is a minimal reproducable example:

```
---
title: MRP
...

::: incremental

- first item
- second item

:::

\onslide<+->

first pause

. . .

second pause
```

I suppose there is no elegant solution here.

**Comment by Jmuccigr** [comment]
@christophehenry Correct. For clarity I should have written above "Don't use `\pause` on slide _anywhere after_ other dynamic content, like an incremental list."

Here's a commented working example with two slides, the first with and the second without a list. Two things to note:

1. On the first slide (with list), the pause _before_ the list can be done with either command, but the one after has to be `\onslide`.
1. For the second slide (w/o list), you have to use `\pause` and not `\onslide<+->` or there won't be a pause.

PS I'm not sure what else qualifies as "dynamic content" besides a list.

```
---
title: MRP
...

::: incremental

first pause

<!-- Comment: the following pause could also be an \onslide. Both work before the list. -->

. . .

- first item
- second item

:::

<!-- Comment: the following pause must be an \onslide. \pause won't work after the list. -->

\onslide<+->

second pause

\onslide<+->

final line on slide 1

-----

Slide 2 pause

<!-- Comment: the following pause must be a \pause. \onslide won't work without a list on the slide. -->

. . .

second line on slide 2

```
